src/metabase/api/public.clj

```diff
@@ -606,6 +606,7 @@
    dashcard-id ms/PositiveInt
    parameters  [:maybe ms/JSONString]}
   (validation/check-public-sharing-enabled)
+  (api/check-404 (t2/select-one-pk :model/Card :id card-id :archived false))
```

Comment:
This check was missing.

See the similar check here for non-pivot-table cards:
https://github.com/metabase/metabase/blob/f52e15255b3b26635605ffa203a1cd6a4e0635e5/src/metabase/api/public.clj#L287